bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,266 @@
+        """Place multiple flags on the board"""
+        board: GameBoard = self.games[ctx.author.id].revealed
+        for x, y in coordinates:
+            if board[y][x] == "hidden":
+                board[y][x] = "flag"
+
+        await self.update_boards(ctx)
+
+    async def lost(self, ctx: commands.Context) -> None:
+        """The player lost the game"""
+        game = self.games[ctx.author.id]
+        game.revealed = game.board
+        await ctx.author.send(":fire: You lost! :fire:")
+        if game.activated_on_server:
+            await game.chat_msg.channel.send(f":fire: {ctx.author.mention} just lost Minesweeper! :fire:")
+
+    async def won(self, ctx: commands.Context) -> None:
+        """The player won the game"""
+        game = self.games[ctx.author.id]
+        game.revealed = game.board
+        await ctx.author.send(":tada: You won! :tada:")
+        if game.activated_on_server:
+            await game.chat_msg.channel.send(f":tada: {ctx.author.mention} just won Minesweeper! :tada:")
+
+    def reveal_zeros(self, revealed: GameBoard, board: GameBoard, x: int, y: int) -> None:
+        """Recursively reveal adjacent cells when a 0 cell is encountered."""
+        for x_, y_ in self.get_neighbours(x, y):
+            if revealed[y_][x_] != "hidden":
+                continue
+            revealed[y_][x_] = board[y_][x_]
+            if board[y_][x_] == 0:
+                self.reveal_zeros(revealed, board, x_, y_)
+
+    async def check_if_won(self, ctx, revealed: GameBoard, board: GameBoard) -> bool:
+        """Checks if a player has won"""
+        if any(
+            revealed[y][x] in ["hidden", "flag"] and board[y][x] != "bomb"
+            for x in range(10)
+            for y in range(10)
+        ):
+            return False
+        else:
+            await self.won(ctx)
+            return True
+
+    async def reveal_one(
+            self,
+            ctx: commands.Context,
+            revealed: GameBoard,
+            board: GameBoard,
+            x: int,
+            y: int
+    ) -> bool:
+        """
+        Reveal one square.
+
+        return is True if the game ended, breaking the loop in `reveal_command` and deleting the game
+        """
+        revealed[y][x] = board[y][x]
+        if board[y][x] == "bomb":
+            await self.lost(ctx)
+            return True
+        elif board[y][x] == 0:
+            self.reveal_zeros(revealed, board, x, y)
+        return await self.check_if_won(ctx, revealed, board)
+
+    @commands.dm_only()
+    @minesweeper_group.command(name="reveal")
+    async def reveal_command(self, ctx: commands.Context, *coordinates: CoordinateConverter) -> None:
+        """Reveal multiple cells"""
+        game = self.games[ctx.author.id]
+        revealed: GameBoard = game.revealed
+        board: GameBoard = game.board
+
+        for x, y in coordinates:
+            # reveal_one returns True if the revealed cell is a bomb or the player won., ending the game
```

Comment:
Did i do that? whops :)
```suggestion
            # reveal_one returns True if the revealed cell is a bomb or the player won, ending the game
```

Reply:
You forgot to mention that's it returns true when they win to, I guess I am just used to putting `.`